integration-tests/dsl-modeline/src/test/java/org/apache/camel/quarkus/component/dsl/modeline/it/DslModelineTest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.dsl.modeline.it;
+
+import java.util.List;
+
+import javax.ws.rs.core.Response;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@QuarkusTest
+public class DslModelineTest {
+
+    @Test
+    void testModelineSingleDependency() {
+        String line = "// camel-k: dependency=mvn:org.my:application:1.0";
```

Comment:
We test that the parsing of the modeline bits work correctly, which is great.

But, does stuff like declaring additional dependencies actually work in a real Camel Quarkus app?

Reply:
I put it to JVM only for now

Reply:
Might be best to tweak the extension metadata so that it's consistent with the other JVM only extensions. E.g add `unlisted: true`. Otherwise it shows up on code.quarkus.io.

Reply:
@jamesnetherton  oh yes nice catch. I regenerated this file. The status was also incorrect. Moved to "Preview", as other JVM extensions